ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package localproject defines LocalProject interface and implements them on localProject
+
+package localproject
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/converter"
+	ecsclient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/ecs"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/secretsmanager"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/urfave/cli"
+)
+
+const (
+	LocalOutDefaultFileName = "./docker-compose.local.yml"
+	LocalOutFileMode        = os.FileMode(0644) // Owner=read/write, Other=readonly
+	LocalInFileName         = "./task-definition.json"
+)
+
+type LocalProject interface {
+	ReadTaskDefinition() error
+	Convert() error
+	Write() error
+	LocalOutFileName() string
+}
+
+type localProject struct {
+	context          *cli.Context
+	taskDefinition   *ecs.TaskDefinition
+	localBytes       []byte
+	localOutFileName string
+}
+
+func New(context *cli.Context) LocalProject {
```

Comment:
I was debating this one. Originally I just had LocalProject as a concrete struct, but then as I looked at the way we have an ecsProject (modeled after libcompose) I started wondering if we may want to extend this interface to take into account, e.g. other versions of compose. However since the actual compose conversion logic lives in the converter package, I could be convinced to switch this back to a concrete struct and remove the LocalProject interface. What do you think?

Reply:
ah I see, I think we can keep it as is, it's not a blocker. 

From your explanation, it does sound like having a public interface defined in this package (for clients to use) but returning a specific type would be the ideal route (denoting the version of compose)